src/main/context-handler.ts

```diff
@@ -91,6 +85,16 @@ export class ContextHandler {
 
   protected async getPrometheusService(): Promise<PrometheusService> {
     const providers = this.listPotentialProviders();
+
+    if (this.preferences.prometheus && providers.length == 1) {
+      return {
+        id: providers[0].id,
+        namespace: this.preferences.prometheus.namespace,
+        service: this.preferences.prometheus.service,
+        port: this.preferences.prometheus.port,
+      };
+    }
```

Comment:
The `id` is the same as `this.prometheusProvider` above so if both are set you don't even need to do the `listPotentialProviders`